contrib/DEVELOPMENT_WORKFLOW.md

```diff
@@ -1,159 +1,328 @@
 # Bittensor Development Workflow
 
-## Table of contents
+## Table
+- [1. Architecture](#1-architecture)
```

Comment:
NACK: this is better suited to docs and is subject to change so should not be included in the Developer Workflow.  This should be restricted to specifics about how to developers should operate while contributing, not educating them on the bittensor network overview.